product_uom_updatable/models/product_template.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 Acsone SA/NV
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from itertools import groupby
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = "product.template"
+
+    def write(self, vals):
+        uom_id = vals.pop("uom_id", False)
+        uom_po_id = vals.pop("uom_po_id", False)
+        res = super(ProductTemplate, self).write(vals)
+        if uom_id:
+            self._update_uom(uom_id)
+        if uom_po_id:
+            self._update_uom_po(uom_po_id)
```

Comment:
Is there a reason for updating the uom *after* the call to super? Intuitively, I would have done it before, so if something in `write` is based on the uom (any computation based on it), it would be correctly using the new one.